grails-app/controllers/org/pih/warehouse/product/ProductSupplierController.groovy

```diff
@@ -283,47 +283,88 @@ class ProductSupplierController {
 
         def productSuppliers = []
 
-        if (params.hasProperty("productSupplier.id")) {
-            productSuppliers = ProductSupplier.findAllByIdInList(params.list("productSupplier.id"))
-        }
-        else {
-            productSuppliers = ProductSupplier.createCriteria().list {
-                resultTransformer(CriteriaSpecification.ALIAS_TO_ENTITY_MAP)
-                projections {
-                    property("active", "active")
-                    property("id", "id")
-                    property("name", "name")
-                    property("code", "code")
-                    product {
-                        property("productCode", "productCode")
-                        property("name", "productName")
-                    }
-                    property("productCode", "legacyProductCode")
-                    supplier(JoinType.LEFT_OUTER_JOIN.joinTypeValue) {
-                        property("name", "supplier.name")
-                    }
-                    property("supplierCode", "supplierCode")
-                    manufacturer(JoinType.LEFT_OUTER_JOIN.joinTypeValue) {
-                        property("name", "manufacturer.name")
-                    }
-                    property("manufacturerCode", "manufacturerCode")
-                    property("minOrderQuantity", "minOrderQuantity")
-                    contractPrice(JoinType.LEFT_OUTER_JOIN.joinTypeValue) {
-                        property("price", "contractPrice.price")
-                        property("toDate", "contractPrice.toDate")
-                    }
-                    defaultProductPackage(JoinType.LEFT_OUTER_JOIN.joinTypeValue) {
-                        uom(JoinType.LEFT_OUTER_JOIN.joinTypeValue) {
-                            property("code", "defaultProductPackage.uom.code")
+        if (params.exportResults) {
+            def command = new ProductSupplierFilterCommand(
+                    searchTerm: params.searchTerm ?: null,
+                    product: params.product ?: null,
+                    includeInactive: params.includeInactive ?: null,
+                    supplier: params.supplier ?: null,
+                    defaultPreferenceTypes: params.defaultPreferenceTypes ? params.list('defaultPreferenceTypes') : null,
+                    createdFrom: params.createdFrom ? new SimpleDateFormat("MM/dd/yyyy").parse(params.createdFrom) : null,
+                    createdTo: params.createdTo ? new SimpleDateFormat("MM/dd/yyyy").parse(params.createdTo) : null,
+                    sort: "dateCreated",
+                    order: "desc",
+                    unlimitedMax: true,
+            )
```

Comment:
this feels odd - command should be bound automatically (use it as an argument of the method, like we used to do), but again - in my opinion this should be a separate Api endpoint instead of refactoring this old one.